rust_src/admin/remacs-helpers.el

```diff
@@ -0,0 +1,34 @@
+;;; Package -- helpers for Remacs porting
+
+(defun remacs-helpers/make-rust-args-from-C-worker (input)
+  "Transform function C arguments INPUT into Rust style arguments."
+  (mapconcat (lambda (arg) (let* ((is-struct (lambda (s) (string= s "struct")))
+                                  (pieces (cl-remove-if is-struct (split-string (string-trim arg) " ")))
+                                  (name (s-append ":" (car (last pieces))))
+                                  (rest (butlast pieces)))
+                             (if (s-starts-with? "\*" name)
+                                 (s-join " " (cons (s-chop-prefix "*" name)
+                                                   (cons "*" rest)))
+                               (s-join " " (cons name rest)))))
+             (split-string input ",")
+             ", "))
```

Comment:
If you're using s.el, might I suggest using dash.el as well? `->>` is lovely, and lets you do things like this:

```emacs-lisp
(defun remacs-helpers/make-rust-args-from-C-worker (input)
  "Transform function C arguments INPUT into Rust style arguments.

For example, convert this:
int *foo, struct Bar b

into this:
foo: * int, b: Bar"
  (->> input
       (s-split ",")
       (-map #'s-trim)
       (--map (s-chop-prefix "struct " it))
       (--map (s-split " " it))
       (-map (-lambda ((type name))
               (if (s-starts-with-p "*" name)
                   (format "%s: * %s" (s-chop-prefix "*" name) type)
                 (format "%s: %s" name type))))
       (s-join ", ")))
```

Reply:
Neat. I was not familiar with dash.